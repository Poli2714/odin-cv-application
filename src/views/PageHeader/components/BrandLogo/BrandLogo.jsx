import styles from './BrandLogo.module.css';
import Logo from 'src/assets/brand-logo.svg';

const size = 50;

function BrandLogo() {
  return (
    <div data-testid="brand-logo" className={styles.brand}>
      <img src={Logo} alt="MEK brand logo" width={size} height={size} />
      <h1 className={styles.brandLogo}>MEK</h1>
    </div>
  );
}

export default BrandLogo;
