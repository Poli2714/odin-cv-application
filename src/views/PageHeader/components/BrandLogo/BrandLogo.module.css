.brand {
  display: flex;
  align-items: center;
  gap: 0.5rem;

  &:hover {
    cursor: pointer;
  }
}

.brandLogo {
  font-size: 2.25rem;
}
