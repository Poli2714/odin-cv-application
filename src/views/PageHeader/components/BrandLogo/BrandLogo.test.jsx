import { test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';

import BrandLogo from './BrandLogo';

test('renders BrandLogo', () => {
  render(<BrandLogo />);

  expect(screen.getByTestId('brand-logo')).toMatchSnapshot();
});

test('renders correct icon inside BrandLogo', () => {
  render(<BrandLogo />);

  expect(screen.getByRole('img').src).toMatch(/\/brand-logo.svg$/i);
});
