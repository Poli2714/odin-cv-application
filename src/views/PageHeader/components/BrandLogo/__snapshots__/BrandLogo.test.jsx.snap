// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders BrandLogo 1`] = `
<div
  class="_brand_536f07"
  data-testid="brand-logo"
>
  <img
    alt="MEK brand logo"
    height="50"
    src="/src/assets/brand-logo.svg"
    width="50"
  />
  <h1
    class="_brandLogo_536f07"
  >
    MEK
  </h1>
</div>
`;
